internal/lsp/server.go

```diff
@@ -429,25 +439,26 @@ func (l *LanguageServer) processTextContentUpdate(
 	return false, nil
 }
 
-func (l *LanguageServer) processBuiltinsUpdate(
-	_ context.Context,
-	uri string,
-	content string,
-) (bool, error) {
+func (l *LanguageServer) processBuiltinsUpdate(_ context.Context, uri string, content string) error {
+	if _, ok := l.cache.GetFileContents(uri); !ok {
+		// If the file is not in the cache, exit early or else
+		// we might accidentally put it in the cache after it's been
+		// deleted: https://github.com/StyraInc/regal/issues/679
```

Comment:
Reading that issue, I have to wonder: does this change also take care of renamed files?

Reply:
Hmm, no, I'll look into it. Thanks!

Reply:
That's now handled 👍